test/WhiteBoxTests.cpp

```diff
@@ -1196,6 +1200,56 @@ void WhiteBoxTests::testThreadPool()
 //    LOK_ASSERT_EQUAL(size_t(7 + existingUnrelatedThreads), waitForThreads(8 + existingUnrelatedThreads));
 }
 
+
+void WhiteBoxTests::testRemoveFile()
+{
+    constexpr std::string_view testname = __func__;
+
+    const std::string baseDir = FileUtil::getSysTempDirectoryPath();
+    const std::string testFile = baseDir + "/removefile_test_file.tmp";
+    const std::string testDir = baseDir + "/removefile_test_dir.tmp";
+    const std::string innerFile = testDir + "/inner_file.txt";
+
+    FileUtil::removeFile(testFile, false);
+    FileUtil::removeFile(testDir, true);
+
+    // test for file
+    {
+        std::ofstream ofs(testFile);
+        ofs << "Test content";
+        ofs.close();
+
+        LOK_ASSERT(FileUtil::Stat(testFile).exists());
+
+        FileUtil::removeFile(testFile, false);
+
+        LOK_ASSERT(!FileUtil::Stat(testFile).exists());
+
+        FileUtil::removeFile(testFile, false);
+        LOK_ASSERT(!FileUtil::Stat(testFile).exists());
+    }
+
+    // test for directory
+    {
+        LOK_ASSERT(FileUtil::makeDirectory(testDir) == 0);
+        std::ofstream ofs(innerFile);
+        ofs << "Inner content";
+        ofs.close();
+
+        LOK_ASSERT(FileUtil::Stat(testDir).exists());
+        LOK_ASSERT(FileUtil::Stat(innerFile).exists());
+
+        FileUtil::removeFile(testDir, true);
```

Comment:
We should test with `false` first. That is, we should test that removing the directory when it has 1 file in it doesn't work and assert that both the directory and `innerFile` exist after `FileUtil::removeFile(testDir, false);`. Then we can remove with `true`.